packages/manager/apps/vrack-services/src/utils/vs-utils.ts

```diff
@@ -157,6 +163,60 @@ export const useUpdateVrackServices = ({
   };
 };
 
+export const useDeleteVrackServices = ({
+  vrackServices,
+  onSuccess,
+  onError,
+}: {
+  vrackServices: string;
+  onSuccess?: () => void;
+  onError?: (result: ApiError) => void;
+}) => {
+  const [isErrorVisible, setIsErrorVisible] = React.useState(false);
+  const queryClient = useQueryClient();
+  const { data: servicesId, isError, error } = useQuery<
+    ApiResponse<number[]>,
+    ApiError
+  >({
+    queryKey: GetVrackServicesServiceIdQueryKey({ vrackServices }),
+    queryFn: () => GetVrackServicesServiceId({ vrackServices }),
+    enabled: !!vrackServices,
+  });
+
+  const {
+    mutate: deleteVs,
+    isError: isTerminateError,
+    error: terminateError,
+    isSuccess,
+  } = useMutation({
+    mutationFn: () =>
+      DeleteVrackServices({
+        serviceId: servicesId?.data[0],
+      }),
+    mutationKey: DeleteVrackServicesQueryKey(vrackServices),
+    onSuccess: () => {
+      queryClient.invalidateQueries({
+        queryKey: DeleteVrackServicesQueryKey(vrackServices),
+      });
+      onSuccess?.();
+    },
+    onError,
+  });
+
+  React.useEffect(() => {
+    if (isError || isTerminateError) {
+      setIsErrorVisible(true);
+    }
+  }, [isError, isTerminateError]);
+
```

Comment:
You don't need to create a state here
You can just return `isErrorVisible: isError || isTerminateError,`

Reply:
In that case, the isErrorVisible won't be updated if isTerminateError is true while executing the mutation no ?